tst/com/amazon/corretto/crypto/provider/test/EvpSignatureSpecificTest.java

```diff
@@ -98,6 +138,48 @@ private static void testKeyTypeMismatch(final String algorithm, final String bas
         }
     }
 
+    @ParameterizedTest
+    @MethodSource("ecdsaBadSignatures")
+    @SuppressWarnings("unchecked")
+    public void testBadEcdsaSignature(final String algorithm, final BigInteger r, final BigInteger s)
+            throws GeneralSecurityException {
+        final Signature verifier = Signature.getInstance(algorithm, NATIVE_PROVIDER);
+        // We always use the largest key to avoid hash truncation confusion
+        final ECPublicKey pubKey = (ECPublicKey) ECDSA_PAIR.getPublic();
+        final byte[] signature;
+        byte[] rArr = r.toByteArray();
+        if (rArr[0] == 0) { // Trim leading zero byte if present
+            rArr = Arrays.copyOfRange(rArr, 1, rArr.length);
+        }
+        byte[] sArr = s.toByteArray();
+        if (sArr[0] == 0) { // Trim leading zero byte if present
+            sArr = Arrays.copyOfRange(sArr, 1, sArr.length);
+        }
+
+        if (algorithm.endsWith("inP1363Format")) {
+            // Just zeros of the appropriate length
+            final int elementLength = (pubKey.getParams().getOrder().bitLength() + 7) / 8;
+            signature = new byte[elementLength * 2];
+            System.arraycopy(rArr, 0, signature, elementLength - rArr.length, rArr.length);
+            System.arraycopy(sArr, 0, signature, 2 * elementLength - sArr.length, sArr.length);
+        } else {
+            // It isn't worth building full ASN.1 logic here, so we cover the easy cases of ones and zeros
+            // and leave the complicated cases of the full orders to the P1363 tests
+            // (which cover the same implementations) under the hood
+            assumeTrue(rArr.length == 1 && sArr.length == 1, "This test only supports {0,1} for non-P1363 sigs.");
+            signature = new byte[]{0x30, 0x06, 0x02, 0x01, 0x00, 0x02, 0x01, 0x00};
```

Comment:
what are these magic values?

Reply:
That's an ASN.1 encoded sequence of two zero values.